lxd/db/cluster/permissions.go

```diff
@@ -109,6 +109,41 @@ func GetPermissionEntityURLs(ctx context.Context, tx *sql.Tx, permissions []Perm
 	return validPermissions, danglingPermissions, entityURLs, nil
 }
 
+// DeletePermissions deletes the given permissions.
+func DeletePermissions(ctx context.Context, tx *sql.Tx, danglingPermissions []Permission) error {
+	if len(danglingPermissions) == 0 {
+		return nil
+	}
+
+	args := make([]any, 0, len(danglingPermissions))
+	for _, perm := range danglingPermissions {
+		args = append(args, perm.ID)
+	}
+
+	var q string
+	if len(args) == 1 {
```

Comment:
This is overly-complex in my view, why not always use the latter form with just a single item?

Reply:
Looking back at this I initially misread the `query.Params` function and thought it didn't handle the single argument case correctly. I'll change it.

Reply:
Done.